client/src/components/DisplayTable/DisplayTableUtils.js

```diff
@@ -0,0 +1,48 @@
+import "./DisplayTableUtils.scss";
+import text from "./DisplayTable.yaml";
+import { create_text_maker_component } from "../misc_util_components.js";
+
+import { WriteToClipboard } from "../WriteToClipboard.js";
+import { IconCopy, IconDownload } from "../../icons/icons.js";
+
+const { text_maker } = create_text_maker_component(text);
+
+export class DisplayTableUtils extends React.Component {
+  downloadCsv() {
+    const { data_to_csv_string, table_name } = this.props;
+    const uri =
+      "data:text/csv;charset=UTF-8," + encodeURIComponent(data_to_csv_string);
+
+    const temporary_anchor = document.createElement("a");
+    temporary_anchor.setAttribute(
+      "download",
+      `${table_name ? table_name : "table"}.csv`
+    );
+    temporary_anchor.setAttribute("href", uri);
+    temporary_anchor.dispatchEvent(new MouseEvent("click"));
+  }
```

Comment:
Please test in IE, been a while but I have a vague memory that this approach might not work there. If it doesn't work in IE then don't spend too long trying to find something that does, we can just hide the download button on IE, the copy to clipboard covers the functionality well enough.

Reply:
Yup, I'll remove it for IE. It doesn't work